### PR TITLE
Add Admin UI (#30)

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**

The admin UI allows setting of user info and permissions. The user info is the user's name, email, job title and their region. Permissions are modeled to be flexible. A user permission is made up of a region and a scope. A scope represents the ability to do something specific, like `READ_REPORTS`. Combined with the region the permission system should be flexible enough to handle user's that need permissions that cross regions.

This is not the final version of the admin UI, although I feel like the UI itself won't be updated much. I expect the schema of the data that drives the admin UI will be updated and will require the UI to be updated as well. But I feel this is a good base for those changes to be applied when needed once the permission model has been finalized, and when we have a more definitive list of scopes.

Out of scope:
 * Any API work, including saving/modeling/REST endpoints
 * Checking/usage of permissions/user info in any other part of the TTAHUB
 * Form validations
 * Final schema for users/permissions. Note global permissions are "region-less" and are represented with "region 0" in this PR. I expect that will change with #50

**How to test**

1) `yarn deps` (`yarn docker:deps`)
2) `yarn start:local` (`yarn docker:start`)
3) Browse to `localhost:3000/admin`

**Issue(s)**
* Closes https://github.com/HHS/Head-Start-TTADP/issues/39

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: frontend/src/components/IndeterminateCheckbox.js
Comment (jasalisbury) [inline frontend/src/components/IndeterminateCheckbox.js:1]: This component was added when the admin UI had a different design and is not currently used. I kept it around because it actually took me a bit of time to figure out and it gives us a nice option for a checkbox in the future. I'm open to removing it if you fell it clutters the codebase too much (it is in the adhoc repos history at this point so wouldn't be too hard to restore it if we end up deleting it)

File: frontend/src/pages/Admin/components/CurrentPermissions.js
Comment (rahearn) [inline frontend/src/pages/Admin/components/CurrentPermissions.js:8]: I don't believe the `key` is needed here, it would be needed on the `<CurrentPermissions>` component, most likely.

**Review by rahearn — COMMENTED** [review]
No code changes needed, but I want to ask about the regions thing to inform #50 and #52

File: frontend/src/components/__tests__/RegionDropdown.js
Comment (jasalisbury) [reply frontend/src/components/__tests__/RegionDropdown.js:8]: I can reword the test, it should say `test('shows the default option when no value is selected')`.

Comment (rahearn) [reply frontend/src/components/__tests__/RegionDropdown.js:8]: What does the default option mean though? Is the control is inactive? Does it correspond to central office?

File: frontend/src/pages/Admin/UserInfo.js
Comment (jasalisbury) [reply frontend/src/pages/Admin/UserInfo.js:29]: I was thinking it would be helpful to know what region the user is "assigned" to in order to default views to that region. If a user typically (always?) works in a specific region (and we know it) we can default views (such as region dropdowns) to that specific region.

I'm still thinking about how to handle the central office users, maybe add a `central office` option?

File: frontend/src/components/__tests__/RegionDropdown.js
Comment (jasalisbury) [reply frontend/src/components/__tests__/RegionDropdown.js:8]: The default option displays `Select an region...` in the UI. It represents a user that hasn't been assigned to a region.

File: frontend/src/pages/Admin/UserInfo.js
Comment (rahearn) [reply frontend/src/pages/Admin/UserInfo.js:29]: If we're going to keep this control, then I think a `central office` option is needed.

I can also think of at least one user who does have roles in multiple regions. And a lot of feds are both `Program Specialists` and `COR` for title. That doesn't mean we can't use these settings as defaults for their normal regions and roles (I like that idea), but it does mean that we can't rely on them for authorization purposes.

Comment (jasalisbury) [reply frontend/src/pages/Admin/UserInfo.js:29]: oh yeah sorry I should have made it clear I wasn't planning on this field being used for authorization or anything other than a helper for the UI.

**Comment by rahearn** [comment]
This will be ready to merge once adhocteam#34 is merged in.

**Review by rahearn — APPROVED** [review]

**Pull request merged**
